app/classes/document_query.rb

```diff
@@ -265,31 +67,99 @@ def timestamp_filters_present?
 
   private
 
-  def child_match(json, field, query, operator = :and)
-    json.child! do
-      json.match do
-        json.set! field do
-          json.operator operator
-          json.query query
-        end
-      end
-    end if query
-  end
-
-  def child_term_filter(json, field, value)
-    json.child! do
-      json.term do
-        json.set! field, value
-      end
-    end
-  end
+  def parse_query(query)
+    site_params_parser = QueryParser.new(query)
+    @site_filters = site_params_parser.site_filters
+    @included_sites = @site_filters[:included_sites]
+    @excluded_sites = @site_filters[:excluded_sites]
+    @query = site_params_parser.stripped_query
+  end
+
+  def set_highlight_options
+    highlight_fields = highlight_fields_hash
+    search.highlight do
+      pre_tags HIGHLIGHT_OPTIONS[:pre_tags]
+      post_tags HIGHLIGHT_OPTIONS[:post_tags]
+      fields highlight_fields
+    end
+  end
+
+  def highlight_fields_hash
+    {
+      ['title',language].compact.join('_') => { number_of_fragments: 0 },
+      ['description',language].compact.join('_') => { fragment_size: 75, number_of_fragments: 2 },
+      ['content',language].compact.join('_') => { fragment_size: 75, number_of_fragments: 2 },
+    }
+  end
+
+  def suggestion_highlight
+    { pre_tag: HIGHLIGHT_OPTIONS[:pre_tags].first,
+      post_tag: HIGHLIGHT_OPTIONS[:post_tags].first }
+  end
+
+  def build_search_query
+    #DSL reference: https://github.com/elastic/elasticsearch-ruby/tree/master/elasticsearch-dsl
+    doc_query = self
```

Comment:
@noremmie , this may raise an eyebrow. I found that the DocumentQuery methods were not available inside the `search...` block. I did some digging to see if there was a more elegant solution (https://www.skorks.com/2013/03/a-closure-is-not-always-a-closure-in-ruby/ included some trickery I considered), but ultimately I decided that this approach, while ugly, was also the cleanest. I'm all ears if you have a better solution.

Reply:
I agree that it's useful to have the full query all in one place per your previous comment. I'm also OK with creating the local variable `doc_query` so that it's accessible from the block.

Perhaps someday we could attach meaningful names to different parts of the query and pull those out as methods, but I'm not asking that we do that right now.

There are some whitespace/indentation issues in the `build_search_query` method that would be good to clean up. It would make it easier for me to read the entire query if things lined up as expected.